viewing_party/party.py

```diff
@@ -1,23 +1,136 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        return {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+            }
+    
+    return None
 
+def add_to_watched(user_data, movie):
+    if movie not in user_data.get("watched"):
+        user_data.get("watched").append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    if movie not in user_data.get("watchlist"):
+        user_data.get("watchlist").append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+    movies = user_data.get("watchlist")
+
+    for movie in movies:
+        if title == movie["title"]:
+            movies.remove(movie)
+            user_data.get("watched").append(movie)
+    
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating = []
+    try:
+        for movie in user_data["watched"]:
+            rating.append(movie["rating"])
+        return sum(rating) / len(rating)
+    except ZeroDivisionError:
+        return 0.0
+    except KeyError: # Incase the key watched is missing
+        return None
 
+def get_most_watched_genre(user_data):
+    genre = {}
 
+    try:
+        for movie in user_data["watched"]:
+            if movie["genre"] not in genre:
+                genre[movie["genre"]] = 1

```

Comment:
Instead of setting it to 1, this should set it to 0 since you immediately add 1 on the next line. The way this is written, all genres get set to '2' immediately after being seen once. Since this happens for all genres, this doesn't impact the functionality, but it is confusing.